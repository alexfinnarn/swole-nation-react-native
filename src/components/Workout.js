import React, { useState } from 'react';
import {Text, View, TextInput, StyleSheet, Button} from "react-native";

export default function Workout({ navigation }) {
  const workout =   {
    id: 1,
    name: 'Stronglifts A',
    description: 'Squats, Overhead Press, Deadlifts'
  };

  const [name, setName] = useState(workout.name);
  const [description, setDescription] = useState(workout.description);

  function handleSave() {

  }

  return (
    <View style={{padding: 20}}>
      <Text style={styles.bold}>Workout Name</Text>
      <TextInput
        style={[
          styles.editText,
          styles.mediumTextInputFont
        ]}
        placeholder="Workout A"
        onChangeText={(text) => setName(text)}
        value={name}
      />
      <Text style={styles.bold}>Workout Description</Text>
      <TextInput
        style={[
          styles.editText,
          styles.smallTextInputFont,
          styles.multiline
        ]}
        multiline
        placeholder="Workout Description"
        onChangeText={(text) => setDescription(text)}
        value={description}
      />
      <Text style={[styles.bold, {marginTop: 15, fontSize: 20}]}>Exercises</Text>
      <View style={{alignItems: 'flex-end', marginTop: 20}}>
        <Button onPress={handleSave} title="Save" />
      </View>

    </View>
  );
}

const styles = StyleSheet.create({
  editText: {
    // alignSelf: 'flex-start',
    color: 'black',
    marginBottom: 20,
    marginTop: 5,
    borderBottomColor: 'grey',
    borderBottomWidth: 2
  },
  mediumTextInputFont: {
    fontSize: 18
  },
  smallTextInputFont: {
    fontSize: 16
  },
  multiline: {
    textAlignVertical: 'top'
  },
  bold: {
    fontWeight: 'bold'
  }
});
