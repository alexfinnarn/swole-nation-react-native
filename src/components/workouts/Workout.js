import React, {useState} from 'react';
import {Text, View, TextInput, Picker} from "react-native";
import WorkoutExerciseListProvider from "../exercises/WorkoutExerciseListProvider";
import {home, styles} from '../Styles';
import AddExercise from "../exercises/AddExercise";
import ActionButton from "../utility/ActionButton";

function Workout({workout, handle, navigation, transformers, actionType}) {
  const [name, setName] = useState(workout.name);
  const [description, setDescription] = useState(workout.description);
  const [transformerKey, setTransformerKey] = useState(transformers[0].key);

  function update(action) {
    let newWorkout = workout;
    newWorkout.name = name;
    newWorkout.description = description;
    handle.update(newWorkout, action, transformerKey);

    if (action === 'CREATE_SESSION') {
      navigation.navigate('Session');
    } else {
      navigation.goBack();
    }
  }

  function ActionButtons(workout) {
    let add = '';
    if (actionType === 'go') {
      add = <ActionButton styles={{marginRight: 5}} key="add" text="Go" action={() => update('CREATE_SESSION')}/>;
    } else {
      add = <ActionButton styles={{marginRight: 5}} key="add" text="Save" action={() => update('UPDATE_WORKOUT')}/>;
    }

    const button = <ActionButton key="button" text="Add Exercise" action={() =>
      navigation.navigate('AddExercise', {pickerEnabled: true, workout: workout})}/>;
    return ([add, button]);
  }

  return (
    <View style={{padding: 10, flex: 1}} testID="workout-root">
      <View style={{flex: 2}}>
        <View style={{flex: 1, flexDirection: 'row', alignItems: 'baseline'}}>
          <Text style={[styles.bold, styles.mediumTextInputFont, {flex: 1}]}>Name</Text>
          <TextInput
            style={[styles.editText, styles.mediumTextInputFont, {flex: 3}]}
            placeholder="Workout A"
            onChangeText={(text) => setName(text)}
            value={name}
          />
        </View>
        <View style={{flex: 1, flexDirection: 'row', alignItems: 'baseline'}}>
          <Text style={[styles.bold, styles.smallTextInputFont, {flex: 1}]}>Description</Text>
          <TextInput
            style={[styles.editText, styles.smallTextInputFont, {flex: 3}]}
            multiline
            placeholder="Workout Description"
            onChangeText={(text) => setDescription(text)}
            value={description}
          />
        </View>
        <View style={{flex: 1, flexDirection: 'row', alignItems: 'baseline', marginBottom: 15}}>
          <Text style={[styles.bold, styles.smallTextInputFont, {flex: 1, paddingBottom: 15}]}>Transformer</Text>
          <Picker
            selectedValue={transformerKey}
            testID="transformer-picker"
            style={{ flex: 3}}
            onValueChange={(value) => setTransformerKey(value)}>
            {transformers.map((tr) => <Picker.Item key={tr.key} label={tr.label} value={tr.key}/>)}
          </Picker>
        </View>
      </View>
      <View style={{flex: 6}}>
        <WorkoutExerciseListProvider workout={workout} navigation={navigation}/>
      </View>
      <View style={{marginTop: 20, flex: 1, flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between'}}>
        <ActionButtons workout={workout}/>
      </View>
    </View>
  );
}

Workout.navigationOptions = {
  title: 'Workout'
};

export default Workout;
