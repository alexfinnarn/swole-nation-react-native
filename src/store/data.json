{
  "workouts": [
    {
      "name": "Stronglifts A",
      "key": "DQkECwYLCQQ",
      "description": "Squats, Bench Press, Barbell Row",
      "exercises": [
        "Squats Warmup"
      ]
    }
  ],
  "exercises": [
    {
      "name": "Squats Warmup",
      "key": "BwEBDwoFBgQ",
      "image": "../../assets/squats.gif",
      "instructions": "Do with a barbell",
      "sets": [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    }
  ],
  "sessions": [
    {
      "name": "new ...date",
      "key": "BwgFDQ8CCg8",
      "workoutKeys": [],
      "exercises": []
    }
  ],
  "tempExerciseList": [],
  "theThing": "",
  "activeWorkoutIndex": 0,
  "activeSessionIndex": 0,
  "activeExerciseIndex": 0
}
