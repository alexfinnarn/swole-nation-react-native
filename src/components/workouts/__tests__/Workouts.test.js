import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';
import data from '../../../store/data';
import main from '../../../store/main';
import Workout from "../Workout";
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import {Picker} from "react-native";
import WorkoutsExerciseList from '../../exercises/WorkoutExerciseList';

let renderer, instance = {};
let workouts = Object.keys(data.workouts).map(key => data.workouts[key]);
const transformers = Object.keys(data.transformers).map((key) => data.transformers[key]);
const handle = {
  update: jest.fn(),
};
const navigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
  getParam: jest.fn(),
};

beforeEach(() => {
  renderer = render(
    (<Provider store={createStore(main)}>
      <Workout workout={workouts[0]} navigation={navigation} handle={handle} transformers={transformers} actionType="save"/>
    </Provider>));
  instance = renderer.getByTestId('workout-root');
}, 0);

describe('<Workout />', () => {
  // @todo Foiled by the extraData param that is always unique. I shouldn't need that data to re-render so I should remove it.
  // it('Renders correctly', () => {
  //   expect(renderer.toJSON()).toMatchSnapshot();
  // });

  it('Displays the name and description of workout and updates text inputs correctly', () => {
    const nameInput = renderer.getByPlaceholder('Workout A');
    const descriptionInput = renderer.getByPlaceholder('Workout Description');

    expect(nameInput.props.value).toBe('Stronglifts A');
    expect(descriptionInput.props.value).toBe('Squats, Bench Press, Barbell Row');

    fireEvent.changeText(nameInput, 'Foo');
    fireEvent.changeText(descriptionInput, 'Bar');

    expect(nameInput.props.value).toBe('Foo');
    expect(descriptionInput.props.value).toBe('Bar');
  });

  it('Selects a Transformer', () => {
    const element = renderer.getByTestId('transformer-picker');
    expect(element.props.selectedValue).toBe('none');

    // Select other workout.
    fireEvent(element, 'valueChange', 'addFive');
    expect(element.props.selectedValue).toBe('addFive');
  });

  it('Displays one WorkoutsExerciseList', () => {
    expect(instance.findAllByType(WorkoutsExerciseList).length).toBe(1);
  });

  it('Displays "Save" or "Go" button based on "go" navigation parameter', () => {
    expect(renderer.queryByText('Save')).not.toBeNull();
    expect(renderer.queryByText('Go')).toBeNull();

    // Trigger re-render with simulated navigation parameter.
    renderer.update(<Provider store={createStore(main)}>
      <Workout workout={workouts[0]} navigation={navigation} handle={handle} transformers={transformers} actionType="go"/>
    </Provider>);

    expect(renderer.queryByText('Save')).toBeNull();
    expect(renderer.queryByText('Go')).not.toBeNull();
  });

  it('Add exercise button navigates to AddExercise screen', () => {
    // For now, just test that the button doesn't show up.
    expect(renderer.queryByText('Add Exercise')).toBeNull();

    // fireEvent(renderer.getByText('Add Exercise'), 'press');
    // expect(navigation.navigate).toHaveBeenCalledTimes(1);
    // expect(navigation.navigate).toHaveBeenCalledWith("AddExercise", {
    //   "pickerEnabled": true,
    //   "workout": {
    //     "workout": {
    //       "description": "Squats, Bench Press, Barbell Row", "exercises": ["Squats Warmup", "Squats", "Bench Press Warmup", "Bench Press", "Barbell Row Warmup", "Barbell Row"],
    //       "key": "DQkECwYLCQQ",
    //       "name": "Stronglifts A"
    //     }
    //   }
    // });
  });
});
