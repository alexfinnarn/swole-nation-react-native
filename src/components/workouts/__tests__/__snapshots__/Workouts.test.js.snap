// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Workout /> Renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 10,
    }
  }
  testID="workout-root"
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "baseline",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontSize": 18,
            },
            Object {
              "flex": 1,
            },
          ]
        }
      >
        Name
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Workout A"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 2,
              "color": "black",
              "marginTop": 5,
            },
            Object {
              "fontSize": 18,
            },
            Object {
              "flex": 3,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value="Stronglifts A"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "baseline",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontSize": 16,
            },
            Object {
              "flex": 1,
            },
          ]
        }
      >
        Description
      </Text>
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Workout Description"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "borderBottomColor": "grey",
              "borderBottomWidth": 2,
              "color": "black",
              "marginTop": 5,
            },
            Object {
              "fontSize": 16,
            },
            Object {
              "flex": 3,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value="Squats, Bench Press, Barbell Row"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "baseline",
          "flex": 1,
          "flexDirection": "row",
          "marginBottom": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontSize": 16,
            },
            Object {
              "flex": 1,
              "paddingBottom": 15,
            },
          ]
        }
      >
        Transformer
      </Text>
      <View
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <RCTPicker
          items={
            Array [
              Object {
                "label": "N/A",
                "textColor": undefined,
                "value": "none",
              },
              Object {
                "label": "Main Exercises (+5 lbs)",
                "textColor": undefined,
                "value": "addFive",
              },
            ]
          }
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectedIndex={0}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
          testID="transformer-picker"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 6,
      }
    }
  >
    <View
      testID="workout-exercise-list-root"
    >
      <RCTScrollView
        data={
          Array [
            Object {
              "image": 1,
              "instructions": "Do with a barbell",
              "key": "BwEBDwoFBgQ",
              "name": "Squats Warmup",
              "sets": Array [
                Object {
                  "reps": 5,
                  "weight": 45,
                },
                Object {
                  "reps": 5,
                  "weight": 45,
                },
                Object {
                  "reps": 5,
                  "weight": 95,
                },
                Object {
                  "reps": 5,
                  "weight": 135,
                },
                Object {
                  "reps": 3,
                  "weight": 185,
                },
                Object {
                  "reps": 2,
                  "weight": 205,
                },
              ],
            },
            Object {
              "image": 1,
              "instructions": "Do with a barbell",
              "key": "Bw0ECwAKBws",
              "name": "Squats",
              "sets": Array [
                Object {
                  "reps": 5,
                  "weight": 230,
                },
                Object {
                  "reps": 5,
                  "weight": 230,
                },
                Object {
                  "reps": 5,
                  "weight": 230,
                },
                Object {
                  "reps": 5,
                  "weight": 230,
                },
                Object {
                  "reps": 5,
                  "weight": 230,
                },
              ],
            },
            Object {
              "image": 1,
              "instructions": "Do with a barbell",
              "key": "BwwBDAIJDgw",
              "name": "Bench Press Warmup",
              "sets": Array [
                Object {
                  "reps": 5,
                  "weight": 45,
                },
                Object {
                  "reps": 5,
                  "weight": 45,
                },
                Object {
                  "reps": 5,
                  "weight": 95,
                },
                Object {
                  "reps": 3,
                  "weight": 135,
                },
              ],
            },
            Object {
              "image": 1,
              "instructions": "Do with a barbell",
              "key": "DgcBCA8BCQo",
              "name": "Bench Press",
              "sets": Array [
                Object {
                  "reps": 5,
                  "weight": 165,
                },
                Object {
                  "reps": 5,
                  "weight": 165,
                },
                Object {
                  "reps": 5,
                  "weight": 165,
                },
                Object {
                  "reps": 5,
                  "weight": 165,
                },
                Object {
                  "reps": 5,
                  "weight": 165,
                },
              ],
            },
            Object {
              "image": 1,
              "instructions": "Do with a barbell",
              "key": "BwYGAw0NDwo",
              "name": "Barbell Row Warmup",
              "sets": Array [
                Object {
                  "reps": 5,
                  "weight": 95,
                },
              ],
            },
            Object {
              "image": 1,
              "instructions": "Do with a barbell",
              "key": "BwQKAwYKAgA",
              "name": "Barbell Row",
              "sets": Array [
                Object {
                  "reps": 5,
                  "weight": 115,
                },
                Object {
                  "reps": 5,
                  "weight": 115,
                },
              ],
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "elevation": 5,
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 4,
                  "marginTop": 4,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 4,
                    "flexDirection": "column",
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "fontWeight": "bold",
                      },
                      Object {
                        "flex": 4,
                        "paddingBottom": 10,
                      },
                    ]
                  }
                >
                  Squats Warmup
                </Text>
                <Text>
                  5x45 lbs, 
                  5x45 lbs, 
                  5x95 lbs, 
                  5x135 lbs, 
                  3x185 lbs, 
                  2x205 lbs
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "column",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#21897E",
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "color": "#ffffff",
                          "fontSize": 20,
                        }
                      }
                    >
                      Edit
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "elevation": 5,
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 4,
                  "marginTop": 4,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 4,
                    "flexDirection": "column",
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "fontWeight": "bold",
                      },
                      Object {
                        "flex": 4,
                        "paddingBottom": 10,
                      },
                    ]
                  }
                >
                  Squats
                </Text>
                <Text>
                  5x230 lbs, 
                  5x230 lbs, 
                  5x230 lbs, 
                  5x230 lbs, 
                  5x230 lbs
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "column",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#21897E",
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "color": "#ffffff",
                          "fontSize": 20,
                        }
                      }
                    >
                      Edit
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "elevation": 5,
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 4,
                  "marginTop": 4,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 4,
                    "flexDirection": "column",
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "fontWeight": "bold",
                      },
                      Object {
                        "flex": 4,
                        "paddingBottom": 10,
                      },
                    ]
                  }
                >
                  Bench Press Warmup
                </Text>
                <Text>
                  5x45 lbs, 
                  5x45 lbs, 
                  5x95 lbs, 
                  3x135 lbs
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "column",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#21897E",
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "color": "#ffffff",
                          "fontSize": 20,
                        }
                      }
                    >
                      Edit
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "elevation": 5,
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 4,
                  "marginTop": 4,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 4,
                    "flexDirection": "column",
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "fontWeight": "bold",
                      },
                      Object {
                        "flex": 4,
                        "paddingBottom": 10,
                      },
                    ]
                  }
                >
                  Bench Press
                </Text>
                <Text>
                  5x165 lbs, 
                  5x165 lbs, 
                  5x165 lbs, 
                  5x165 lbs, 
                  5x165 lbs
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "column",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#21897E",
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "color": "#ffffff",
                          "fontSize": 20,
                        }
                      }
                    >
                      Edit
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "elevation": 5,
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 4,
                  "marginTop": 4,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 4,
                    "flexDirection": "column",
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "fontWeight": "bold",
                      },
                      Object {
                        "flex": 4,
                        "paddingBottom": 10,
                      },
                    ]
                  }
                >
                  Barbell Row Warmup
                </Text>
                <Text>
                  5x95 lbs
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "column",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#21897E",
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "color": "#ffffff",
                          "fontSize": 20,
                        }
                      }
                    >
                      Edit
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "elevation": 5,
                  "flex": 1,
                  "flexDirection": "row",
                  "marginBottom": 4,
                  "marginTop": 4,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.25,
                  "shadowRadius": 3.84,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 4,
                    "flexDirection": "column",
                    "padding": 10,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "fontWeight": "bold",
                      },
                      Object {
                        "flex": 4,
                        "paddingBottom": 10,
                      },
                    ]
                  }
                >
                  Barbell Row
                </Text>
                <Text>
                  5x115 lbs, 
                  5x115 lbs
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "column",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#21897E",
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "color": "#ffffff",
                          "fontSize": 20,
                        }
                      }
                    >
                      Edit
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
